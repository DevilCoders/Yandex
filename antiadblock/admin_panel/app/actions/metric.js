import serviceApi from 'app/api/service';

export const START_METRIC_LOADING = 'START_METRIC_LOADING';
export const END_METRIC_LOADING = 'END_METRIC_LOADING';

export function fetchMetric(serviceId, name, range) {
    return dispatch => {
        dispatch(startMetricLoading(serviceId, name, range));

        return serviceApi.fetchServiceMetric(serviceId, name, range).then(data => {
            return dispatch(endMetricLoading(serviceId, name, range, data, true));
        }, () => {
            return dispatch(endMetricLoading(serviceId, name, range, []));
        });
    };
}

export function startMetricLoading(serviceId, name, range) {
    return {
        type: START_METRIC_LOADING,
        serviceId,
        name,
        range
    };
}

export function endMetricLoading(serviceId, name, range, data, loaded) {
    return {
        type: END_METRIC_LOADING,
        serviceId,
        name,
        range,
        data,
        loaded
    };
}
