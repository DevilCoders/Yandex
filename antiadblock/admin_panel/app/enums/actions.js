export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
