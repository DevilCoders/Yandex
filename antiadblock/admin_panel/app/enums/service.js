export const STATUS = {
    ACTIVE: 'ok',
    INACTIVE: 'inactive'
};

export const STATUS_COLORS = {
    [STATUS.ACTIVE]: 'green',
    [STATUS.INACTIVE]: 'gray'
};

export const SUPPORTT_PRIORITY = {
    CRITICAL: 'critical',
    MAJOR: 'major',
    MINOR: 'minor',
    OTHER: 'other'
};
