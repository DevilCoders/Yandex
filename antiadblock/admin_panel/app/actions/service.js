import serviceApi from 'app/api/service';
import configApi from 'app/api/config';

import {setGlobalErrors} from 'app/actions/errors';

export const START_SERVICE_LOADING = 'START_SERVICE_LOADING';
export const END_SERVICE_LOADING = 'END_SERVICE_LOADING';
export const START_SERVICE_TREND_LOADING = 'START_SERVICE_TREND_LOADING';
export const END_SERVICE_TREND_LOADING = 'END_SERVICE_TREND_LOADING';
export const START_SERVICE_TREND_CREATION = 'START_SERVICE_TREND_CREATION';
export const END_SERVICE_TREND_CREATION = 'END_SERVICE_TREND_CREATION';
export const START_LABELS_LOADING = 'START_LABELS_LOADING';
export const END_LABELS_LOADING = 'END_LABELS_LOADING';
export const CHANGE_LABEL_PARENT_START = 'CHANGE_LABEL_PARENT_START';
export const CHANGE_LABEL_PARENT_END = 'CHANGE_LABEL_PARENT_END';
export const START_SERVICE_CONFIG_LOADING = 'START_SERVICE_CONFIG_LOADING';
export const END_SERVICE_CONFIG_LOADING = 'END_SERVICE_CONFIG_LOADING';
export const START_SERVICE_CONFIGS_LOADING = 'START_SERVICE_CONFIGS_LOADING';
export const END_SERVICE_CONFIGS_LOADING = 'END_SERVICE_CONFIGS_LOADING';
export const START_SERVICE_CHANGE_PRIORITY = 'START_SERVICE_CHANGE_PRIORITY';
export const END_SERVICE_CHANGE_PRIORITY = 'END_SERVICE_CHANGE_PRIORITY';
export const START_SERVICE_CONFIG_APPLYING = 'START_SERVICE_CONFIG_APPLYING';
export const END_SERVICE_CONFIG_APPLYING = 'END_SERVICE_CONFIG_APPLYING';
export const START_SERVICE_EXPERIMENT_APPLYING = 'START_SERVICE_EXPERIMENT_APPLYING';
export const END_SERVICE_EXPERIMENT_APPLYING = 'END_SERVICE_EXPERIMENT_APPLYING';
export const START_CONFIG_REMOVE_EXPERIMENT = 'START_CONFIG_REMOVE_EXPERIMENT';
export const END_CONFIG_REMOVE_EXPERIMENT = 'END_CONFIG_REMOVE_EXPERIMENT';
export const START_SERVICE_CONFIG_SAVING = 'START_SERVICE_CONFIG_SAVING';
export const END_SERVICE_CONFIG_SAVING = 'END_SERVICE_CONFIG_SAVING';
export const START_SERVICE_SETUP_LOADING = 'START_SERVICE_SETUP_LOADING';
export const END_SERVICE_SETUP_LOADING = 'END_SERVICE_SETUP_LOADING';
export const RESET_SERVICE_AUDIT = 'RESET_SERVICE_AUDIT';
export const START_SERVICE_AUDIT_LOADING = 'START_SERVICE_AUDIT_LOADING';
export const END_SERVICE_AUDIT_LOADING = 'END_SERVICE_AUDIT_LOADING';
export const START_SERVICE_CONFIG_ARCHIVED_SETTING = 'START_SERVICE_CONFIG_ARCHIVED_SETTING';
export const END_SERVICE_CONFIG_ARCHIVED_SETTING = 'END_SERVICE_CONFIG_ARCHIVED_SETTING';
export const START_SERVICE_DISABLING = 'START_SERVICE_DISABLING';
export const END_SERVICE_DISABLING = 'END_SERVICE_DISABLING';
export const START_SERVICE_ENABLING = 'START_SERVICE_ENABLING';
export const END_SERVICE_ENABLING = 'END_SERVICE_ENABLING';
export const START_SERVICE_CONFIG_MODERATING = 'START_SERVICE_CONFIG_MODERATING';
export const END_SERVICE_CONFIG_MODERATING = 'END_SERVICE_CONFIG_MODERATING';
export const START_SERVICE_COMMENT = 'START_SERVICE_COMMENT';
export const END_SERVICE_COMMENT = 'END_SERVICE_COMMENT';
export const START_SET_SERVICE_ENABLE_MONITORING = 'START_SET_SERVICE_ENABLE_MONITORING';
export const END_SET_SERVICE_ENABLE_MONITORING = 'END_SET_SERVICE_ENABLE_MONITORING';
export const START_PARENT_EXP_CONFIGS_LOADING = 'START_PARENT_EXP_CONFIGS_LOADING';
export const END_PARENT_EXP_CONFIGS_LOADING = 'END_PARENT_EXP_CONFIGS_LOADING';

export function fetchService(serviceId) {
    return dispatch => {
        dispatch(startServiceLoading(serviceId));

        return Promise.all(
            [
                serviceApi.fetchService(serviceId),
                serviceApi.fetchLabels(serviceId)
            ])
            .then(([service, labels]) => {
                dispatch(endServiceLoading(serviceId, service));
                dispatch(endLabelsLoading(serviceId, labels));
            }, error => {
                return dispatch(setGlobalErrors([error.message]));
            });
    };
}

export function startServiceLoading(serviceId) {
    return {
        type: START_SERVICE_LOADING,
        serviceId
    };
}

export function endServiceLoading(serviceId, service) {
    return {
        type: END_SERVICE_LOADING,
        serviceId,
        service
    };
}

export function fetchTrend(serviceId) {
    return dispatch => {
        dispatch(startTrendLoading(serviceId));

        return serviceApi.fetchTrend(serviceId)
            .then(trend => {
                dispatch(endTrendLoading(serviceId, trend, true));
            }, error => {
                // Ошибка Not Found, значит тикета нет
                if (error.status === 404) {
                    return dispatch(endTrendLoading(serviceId, {
                        ticket_id: null
                    }, true));
                }

                // Ошибка Wrong Component, значит сервис не заведен в startrek и создать тикет невозможно
                if (error.status === 400) {
                    return dispatch(endTrendLoading(serviceId, {
                        ticket_id: null
                    }, false));
                }

                // Другие ошибки
                return dispatch(setGlobalErrors([error.message]));
            });
    };
}

export function startTrendLoading(serviceId) {
    return {
        type: START_SERVICE_TREND_LOADING,
        serviceId
    };
}

export function endTrendLoading(serviceId, trend, loaded) {
    return {
        type: END_SERVICE_TREND_LOADING,
        serviceId,
        trend,
        loaded
    };
}

export function createTrend(serviceId, datetime) {
    return dispatch => {
        dispatch(startTrendCreation(serviceId, datetime));

        return serviceApi.createTrend(serviceId, datetime)
            .then(trend => {
                dispatch(endTrendCreation(serviceId, trend));
            }, error => {
                return dispatch(setGlobalErrors([error.message]));
            });
    };
}

export function startTrendCreation(serviceId, datetime) {
    return {
        type: START_SERVICE_TREND_CREATION,
        serviceId,
        datetime
    };
}

export function endTrendCreation(serviceId, trend) {
    return {
        type: END_SERVICE_TREND_CREATION,
        serviceId,
        trend
    };
}

export function fetchLabels(serviceId) {
    return dispatch => {
        dispatch(startLabelsLoading(serviceId));

        return serviceApi.fetchLabels(serviceId)
            .then(labels => {
                dispatch(endLabelsLoading(serviceId, labels));
            }, error => {
                return dispatch(setGlobalErrors([error.message]));
            });
    };
}

export function startLabelsLoading(serviceId) {
    return {
        type: START_LABELS_LOADING,
        serviceId
    };
}

export function endLabelsLoading(serviceId, labels) {
    return {
        type: END_LABELS_LOADING,
        serviceId,
        labels
    };
}

export function changeLabelParent(labelId, parentId, serviceId) {
    return dispatch => {
        dispatch(changeLabelParentStart(labelId, parentId, serviceId));

        return serviceApi.changeLabelParent(labelId, parentId)
            .then(() => dispatch(fetchLabels(serviceId)))
            .catch(error => dispatch(setGlobalErrors([error.message])));
    };
}

export function changeLabelParentStart(labelId, parentId, serviceId) {
    return {
        type: CHANGE_LABEL_PARENT_START,
        labelId,
        parentId,
        serviceId
    };
}

export function changeLabelParentEnd(labelId, parentId, serviceId) {
    return {
        type: CHANGE_LABEL_PARENT_END,
        labelId,
        parentId,
        serviceId
    };
}

export function serviceChangePriority(serviceId, priority) {
    return dispatch => {
        dispatch(startServiceChangePriority(serviceId, priority));

        return serviceApi.changePriority(serviceId, priority).then(() => {
            return dispatch(endServiceChangePriority(serviceId, priority));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceChangePriority(serviceId, priority) {
    return {
        type: START_SERVICE_CHANGE_PRIORITY,
        serviceId,
        priority
    };
}

export function endServiceChangePriority(serviceId, priority) {
    return {
        type: END_SERVICE_CHANGE_PRIORITY,
        serviceId,
        priority
    };
}

export function fetchConfig(serviceId, configId) {
    return dispatch => {
        dispatch(startServiceConfigLoading(serviceId, configId));

        return configApi.fetchConfig(configId).then(config => {
            return dispatch(endServiceConfigLoading(serviceId, configId, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceConfigLoading(serviceId, configId) {
    return {
        type: START_SERVICE_CONFIG_LOADING,
        serviceId,
        configId
    };
}

export function endServiceConfigLoading(serviceId, configId, config) {
    return {
        type: END_SERVICE_CONFIG_LOADING,
        serviceId,
        configId,
        config
    };
}

export function fetchServiceConfigs(labelId, offset, limit, filters) {
    return dispatch => {
        dispatch(startServiceConfigsLoading(labelId, offset, limit));

        return serviceApi.fetchServiceConfigs(labelId, offset, limit, filters).then(configs => {
            return dispatch(endServiceConfigsLoading(labelId, configs));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceConfigsLoading(labelId, offset, limit) {
    return {
        type: START_SERVICE_CONFIGS_LOADING,
        labelId,
        offset,
        limit
    };
}

export function endServiceConfigsLoading(labelId, configs) {
    return {
        type: END_SERVICE_CONFIGS_LOADING,
        labelId,
        configs
    };
}

export function applyConfig(labelId, configId, options) {
    return dispatch => {
        dispatch(startServiceConfigApplying(labelId, configId, options));

        return configApi.applyConfig(labelId, configId, options).then(() => {
            return dispatch(endServiceConfigApplying(labelId, configId, options));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceConfigApplying(labelId, configId, options) {
    return {
        type: START_SERVICE_CONFIG_APPLYING,
        labelId,
        configId,
        options
    };
}

export function endServiceConfigApplying(labelId, configId, options) {
    return {
        type: END_SERVICE_CONFIG_APPLYING,
        labelId,
        configId,
        options
    };
}

export function applyExperiment(serviceId, configId, options) {
    return dispatch => {
        dispatch(startServiceExperimentApplying(serviceId, configId, options));

        return serviceApi.applyExperiment(configId, options)
            .then(() => {
                return dispatch(endServiceExperimentApplying(serviceId, configId, options, true));
            }, error => {
                dispatch(endServiceExperimentApplying(serviceId, configId, options, false));
                throw error;
            });
    };
}

export function startServiceExperimentApplying(serviceId, configId, options) {
    return {
        type: START_SERVICE_EXPERIMENT_APPLYING,
        serviceId,
        configId,
        options
    };
}

export function endServiceExperimentApplying(serviceId, configId, options, success) {
    return {
        type: END_SERVICE_EXPERIMENT_APPLYING,
        serviceId,
        configId,
        options,
        success
    };
}

export function removeExperiment(serviceId, configId) {
    return dispatch => {
        dispatch(startConfigRemoveExperiment(serviceId, configId));

        return serviceApi.removeExperiment(configId)
            .then(() => {
                return dispatch(endConfigRemoveExperiment(serviceId, configId));
            }, error => {
                return dispatch(setGlobalErrors([error.message]));
            });
    };
}

export function startConfigRemoveExperiment(serviceId, configId) {
    return {
        type: START_CONFIG_REMOVE_EXPERIMENT,
        serviceId,
        configId
    };
}

export function endConfigRemoveExperiment(serviceId, configId) {
    return {
        type: END_CONFIG_REMOVE_EXPERIMENT,
        serviceId,
        configId
    };
}

export function fetchServiceSetup(serviceId) {
    return dispatch => {
        dispatch(startServiceSetupLoading(serviceId));

        /* TODO: вернуть этот код после отрывания мока
        return serviceApi.fetchServiceSetup(serviceId).then(service => {
            return dispatch(endServiceSetupLoading(serviceId, service));
        }, error => {
            redirect(antiadbUrl(`/error/${error.status}`));
        }); */

        return Promise.resolve(dispatch(endServiceSetupLoading(serviceId, {
            token: 'AABBCCDDEEFF....',
            js_inline: '(func () { /* here goes awesome JS code */})("<PARTNER-ID>")',
            nginx_config: 'here \n goes \n config'
        })));
    };
}

export function startServiceSetupLoading(serviceId) {
    return {
        type: START_SERVICE_SETUP_LOADING,
        serviceId
    };
}

export function endServiceSetupLoading(serviceId, setup) {
    return {
        type: END_SERVICE_SETUP_LOADING,
        serviceId,
        setup
    };
}

export function resetServiceAudit(serviceId) {
    return {
        type: RESET_SERVICE_AUDIT,
        serviceId
    };
}

export function fetchServiceAudit(serviceId, offset, limit, labelId) {
    return dispatch => {
        dispatch(startServiceAuditLoading(serviceId, offset, limit));

        // если есть labelId, то присылаются события только для него.
        return serviceApi.fetchServiceAudit(serviceId, offset, limit, labelId).then(audit => {
            return dispatch(endServiceAuditLoading(serviceId, audit));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceAuditLoading(serviceId, offset, limit) {
    return {
        type: START_SERVICE_AUDIT_LOADING,
        serviceId,
        offset,
        limit
    };
}

export function endServiceAuditLoading(serviceId, audit) {
    return {
        type: END_SERVICE_AUDIT_LOADING,
        serviceId,
        audit
    };
}

export function setArchivedConfig(serviceId, configId, archived) {
    return dispatch => {
        dispatch(startServiceConfigArchivedSetting(serviceId, configId, archived));

        return configApi.setArchivedConfig(configId, archived).then(() => {
            return dispatch(endServiceConfigArchivedSetting(serviceId, configId, archived));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceConfigArchivedSetting(serviceId, configId, archived) {
    return {
        type: START_SERVICE_CONFIG_ARCHIVED_SETTING,
        serviceId,
        configId,
        archived
    };
}

export function endServiceConfigArchivedSetting(serviceId, configId, archived) {
    return {
        type: END_SERVICE_CONFIG_ARCHIVED_SETTING,
        serviceId,
        configId,
        archived
    };
}

export function disableService(serviceId) {
    return dispatch => {
        dispatch(startServiceDisabling(serviceId));

        return serviceApi.disableService(serviceId).then(() => {
            return dispatch(endServiceDisabling(serviceId));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function enableService(serviceId) {
    return dispatch => {
        dispatch(startServiceEnabling(serviceId));

        return serviceApi.enableService(serviceId).then(() => {
            return dispatch(endServiceEnabling(serviceId));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceDisabling(serviceId) {
    return {
        type: START_SERVICE_DISABLING,
        serviceId
    };
}

export function endServiceDisabling(serviceId) {
    return {
        type: END_SERVICE_DISABLING,
        serviceId
    };
}

export function startServiceEnabling(serviceId) {
    return {
        type: START_SERVICE_ENABLING,
        serviceId
    };
}

export function endServiceEnabling(serviceId) {
    return {
        type: END_SERVICE_ENABLING,
        serviceId
    };
}

export function setModerateConfig(serviceId, labelId, configId, approved, comment) {
    return dispatch => {
        dispatch(startModerateServiceConfig(serviceId, configId, approved, comment));

        return configApi.setModerateConfig(labelId, configId, approved, comment).then(config => {
            return dispatch(endModerateServiceConfig(serviceId, configId, approved, comment, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startModerateServiceConfig(serviceId, configId, approved, comment) {
    return {
        type: START_SERVICE_CONFIG_MODERATING,
        serviceId,
        configId,
        approved,
        comment
    };
}

export function endModerateServiceConfig(serviceId, configId, approved, comment, config) {
    return {
        type: END_SERVICE_CONFIG_MODERATING,
        serviceId,
        configId,
        approved,
        comment,
        config
    };
}

export function addServiceComment(serviceId, comment) {
    return dispatch => {
        dispatch(startServiceComment(serviceId));

        return serviceApi.addServiceComment(serviceId, comment).then(() => {
            return dispatch(endServiceComment(serviceId, comment));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startServiceComment(serviceId) {
    return {
        type: START_SERVICE_COMMENT,
        serviceId
    };
}

export function endServiceComment(serviceId, comment) {
    return {
        type: END_SERVICE_COMMENT,
        serviceId,
        comment
    };
}

export function setServiceEnableMonitoring(serviceId, monitoringName, monitoringIsEnable) {
    return dispatch => {
        dispatch(startSetServiceEnableMonitoring(serviceId));

        return serviceApi.setServiceEnableMonitoring(serviceId, monitoringName, monitoringIsEnable).then(() => {
            return dispatch(endSetServiceEnableMonitoring(serviceId, monitoringName, monitoringIsEnable));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startSetServiceEnableMonitoring(serviceId) {
    return {
        type: START_SET_SERVICE_ENABLE_MONITORING,
        serviceId
    };
}

export function endSetServiceEnableMonitoring(serviceId, monitoringName, monitoringIsEnable) {
    return {
        type: END_SET_SERVICE_ENABLE_MONITORING,
        serviceId,
        monitoringName,
        monitoringIsEnable
    };
}

export function fetchParentExpConfigs(labelId) {
    return dispatch => {
        dispatch(startParentExpConfigsLoading(labelId));

        return configApi.fetchExpConfigsId(labelId).then(data => {
            return dispatch(endParentExpConfigsLoading(labelId, data));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startParentExpConfigsLoading(labelId) {
    return {
        type: START_PARENT_EXP_CONFIGS_LOADING,
        labelId
    };
}

export function endParentExpConfigsLoading(labelId, data) {
    return {
        type: END_PARENT_EXP_CONFIGS_LOADING,
        labelId,
        data
    };
}
