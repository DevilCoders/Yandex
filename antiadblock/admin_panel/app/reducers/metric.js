import * as serviceActions from 'app/actions/service';
import * as metricActions from 'app/actions/metric';
import update from 'app/lib/update';

import {METRICS} from 'app/enums/graphs';

export default function(state, action) {
    const {type} = action;

    state = state || {};

    switch (type) {
        // При загрузке сервиса инициализируем state мониторинга пустыми объектами
        case serviceActions.END_SERVICE_LOADING: {
            return update(state, {
                $merge: {
                    [action.service.id]: {
                        ...Object.values(METRICS).reduce((acc, item) => {
                            acc[item] = {
                                loading: false,
                                loaded: false
                            };

                            return acc;
                        }, {})
                    }
                }
            });
        }
        case metricActions.START_METRIC_LOADING: {
            return update(state, {
                $merge: {
                    [action.serviceId]: {
                        ...state[action.serviceId],
                        [action.name]: {
                            range: action.range,
                            loaded: false,
                            loading: true
                        }
                    }
                }
            });
        }
        case metricActions.END_METRIC_LOADING: {
            if (state[action.serviceId][action.name].range === action.range) {
                return update(state, {
                    $merge: {
                        [action.serviceId]: {
                            ...state[action.serviceId],
                            [action.name]: {
                                data: action.data,
                                loading: false,
                                loaded: action.loaded
                            }
                        }
                    }
                });
            }

            return state;
        }
        default: {
            return state;
        }
    }
}

