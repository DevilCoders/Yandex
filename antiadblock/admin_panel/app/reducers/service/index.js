import {combineReducers} from 'redux';

import service from './service';
import audit from './audit';
import configs from './configs';
import applyingConfig from './applying-config';
import moderatingConfig from './moderating-config';
import experimentApplying from './experiment-applying';
import setup from './setup';
import labels from './labels';
import schema from './schema';
import health from './health';
import rules from './rules';
import sbs from './sbs';
import trend from './trend';

const serviceReducer = combineReducers({
    service,
    audit,
    configs,
    applyingConfig,
    moderatingConfig,
    experimentApplying,
    setup,
    schema,
    health,
    rules,
    labels,
    sbs,
    trend
});

export default serviceReducer;

const defaultConfig = {
    items: [],
    loaded: false,
    total: 0
};

export function getActiveLabelId(state) {
    return state.configs.activeLabelId;
}

export function getConfigsByLabelId(state, labelId) {
    return state.configs.byLabelId[labelId] || defaultConfig;
}

export function getConfigs(state) {
    return state.configs;
}

export function getConfigById(state, id) {
    return state.configs.byId[id];
}

export function getAudit(state) {
    return state.audit;
}

export function getAuditItemById(state, id) {
    return state.audit.byId[id];
}

export function getSchema(state) {
    return state.schema;
}

export function getHealth(state) {
    return state.health;
}

export function getSbs(state) {
    return state.sbs;
}

export function getSbsScreenshotsChecks(state) {
    return state.sbs.screenshotsChecks;
}

export function getSbsScreenshotsChecksDataByRunId(state, runId) {
    return state.sbs.screenshotsChecks.runIdData[runId];
}

export function getSbsProfiles(state) {
    return state.sbs.allSbsProfiles;
}

export function getSbsProfile(state) {
    return state.sbs.allSbsProfiles.profile;
}

export function getSbsProfileById(state, id) {
    return (state.sbs.allSbsProfiles.byId[id] ? state.sbs.allSbsProfiles.byId[id] : state.sbs.allSbsProfiles.byId[-1]);
}

export function getSbsProfilesTags(state) {
    return state.sbs.allSbsProfiles.profileTags;
}

export function getSbsResultChecks(state) {
    return state.sbs.resultChecks;
}

export function getRules(state) {
    return state.rules;
}

export function getLabels(state) {
    return state.labels;
}

export function getTrend(state) {
    return state.trend;
}

export function getParentConfigsExpId(state) {
    return state.configs.expIds;
}

export function getSbsScreenshotsChecksGroupVisibility(state) {
    return state.sbs.screenshotsChecks.logsVisible;
}

export function getSbsLogs(state) {
    return state.sbs.logs;
}
