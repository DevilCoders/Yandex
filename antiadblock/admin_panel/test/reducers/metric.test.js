import reducer from 'app/reducers/metric';
import * as metricActionFixtures from '../fixtures/actions/metric';
import * as serviceActionFixtures from '../fixtures/actions/service';
import {METRIC} from '../fixtures/store';

describe('metric', () => {
    const initialState = {};

    describe('reducer', () => {
        it('should return the initial state', () => {
            expect(reducer(undefined, {})).toMatchSnapshot();
        });

        it('should handle START_METRIC_LOADING', () => {
            expect(
                reducer(initialState, metricActionFixtures.START_METRIC_LOADING)
            ).toMatchSnapshot();
        });

        it('should handle END_METRIC_LOADING', () => {
            const afterMetricLoadingStarted = reducer(initialState, metricActionFixtures.START_METRIC_LOADING);
            const afterMetricLoaded = reducer(afterMetricLoadingStarted, metricActionFixtures.END_METRIC_LOADING);

            expect(afterMetricLoaded).toMatchSnapshot();
        });

        it('should handle END_SERVICE_LOADING', () => {
            const afterServiceConfigsLoaded = reducer(METRIC, serviceActionFixtures.END_SERVICE_LOADING);

            expect(afterServiceConfigsLoaded).toMatchSnapshot();
        });

        it('should handle START_METRIC_LOADING and remove data from store', () => {
            const afterServiceConfigsLoaded = reducer(METRIC, metricActionFixtures.START_METRIC_LOADING);

            expect(afterServiceConfigsLoaded).toMatchSnapshot();
        });

        it('should handle END_SERVICE_LOADING and store should not changed', () => {
            const afterMetricLoaded = reducer(METRIC, metricActionFixtures.END_METRIC_LOADING);

            expect(afterMetricLoaded).toMatchSnapshot();
        });
    });
});
