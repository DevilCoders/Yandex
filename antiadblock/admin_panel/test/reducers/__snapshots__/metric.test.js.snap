// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metric reducer should handle END_METRIC_LOADING 1`] = `
Object {
  "test.service": Object {
    "http_codes": Object {
      "data": Object {
        "1524758640000": Object {
          "200": 426,
          "204": 8,
          "301": 40,
          "302": 3,
          "404": 40,
        },
        "1524758700000": Object {
          "200": 2566,
          "204": 75,
          "301": 239,
          "302": 4,
          "304": 4,
          "404": 227,
        },
        "1524758760000": Object {
          "200": 2365,
          "204": 66,
          "301": 270,
          "302": 7,
          "304": 3,
          "404": 247,
        },
        "1524758820000": Object {
          "200": 2624,
          "204": 80,
          "301": 281,
          "302": 12,
          "304": 20,
          "404": 240,
          "502": 1,
        },
        "1524758880000": Object {
          "200": 3368,
          "204": 109,
          "301": 330,
          "302": 13,
          "304": 4,
          "404": 293,
          "502": 1,
        },
        "1524758940000": Object {
          "200": 2359,
          "204": 75,
          "301": 225,
          "302": 13,
          "404": 193,
        },
        "1524759000000": Object {
          "200": 2454,
          "204": 93,
          "301": 197,
          "302": 15,
          "404": 170,
        },
        "1524759060000": Object {
          "200": 2805,
          "204": 92,
          "301": 265,
          "302": 13,
          "304": 2,
          "404": 233,
        },
        "1524759120000": Object {
          "200": 2806,
          "204": 87,
          "301": 230,
          "302": 12,
          "304": 6,
          "404": 213,
        },
        "1524759180000": Object {
          "200": 2368,
          "204": 85,
          "301": 182,
          "302": 11,
          "304": 1,
          "404": 173,
        },
        "1524759240000": Object {
          "200": 2797,
          "204": 100,
          "301": 232,
          "302": 20,
          "304": 7,
          "404": 206,
        },
        "1524759300000": Object {
          "200": 2421,
          "204": 80,
          "301": 214,
          "302": 14,
          "304": 3,
          "404": 197,
        },
        "1524759360000": Object {
          "200": 2311,
          "204": 80,
          "301": 194,
          "302": 13,
          "304": 6,
          "404": 183,
        },
        "1524759420000": Object {
          "200": 2471,
          "204": 86,
          "301": 277,
          "302": 11,
          "304": 3,
          "404": 264,
        },
        "1524759480000": Object {
          "200": 3478,
          "204": 119,
          "301": 378,
          "302": 22,
          "304": 3,
          "404": 355,
        },
        "1524759540000": Object {
          "200": 2321,
          "204": 77,
          "301": 212,
          "302": 11,
          "404": 198,
        },
      },
      "loaded": true,
      "loading": false,
    },
  },
}
`;

exports[`metric reducer should handle END_SERVICE_LOADING 1`] = `
Object {
  "123": Object {
    "adblock_apps_proportions": Object {
      "loaded": false,
      "loading": false,
    },
    "bamboozled_by_app": Object {
      "loaded": false,
      "loading": false,
    },
    "bamboozled_by_bro": Object {
      "loaded": false,
      "loading": false,
    },
    "fetch_timings": Object {
      "loaded": false,
      "loading": false,
    },
    "http_codes": Object {
      "loaded": false,
      "loading": false,
    },
    "http_errors_domain_type": Object {
      "loaded": false,
      "loading": false,
    },
  },
  "test.second.service": Object {
    "bamboozled_by_app": Object {
      "loaded": false,
      "loading": true,
      "range": 330,
    },
    "bamboozled_by_bro": Object {
      "loaded": false,
      "loading": false,
      "range": 330,
    },
    "http_codes": Object {
      "data": Object {
        "1524759180000": Object {
          "200": 2368,
          "204": 85,
          "301": 182,
        },
        "1524759240000": Object {
          "200": 2797,
          "204": 100,
          "301": 232,
        },
      },
      "loaded": true,
      "loading": false,
      "range": 330,
    },
  },
  "test.service": Object {
    "http_codes": Object {
      "data": Object {
        "1524759180000": Object {
          "200": 2368,
          "204": 85,
          "301": 182,
          "302": 11,
          "304": 1,
          "404": 173,
        },
        "1524759240000": Object {
          "200": 2797,
          "204": 100,
          "301": 232,
          "302": 20,
          "304": 7,
          "404": 206,
        },
      },
      "loaded": true,
      "loading": false,
      "range": 10,
    },
  },
}
`;

exports[`metric reducer should handle END_SERVICE_LOADING and store should not changed 1`] = `
Object {
  "test.second.service": Object {
    "bamboozled_by_app": Object {
      "loaded": false,
      "loading": true,
      "range": 330,
    },
    "bamboozled_by_bro": Object {
      "loaded": false,
      "loading": false,
      "range": 330,
    },
    "http_codes": Object {
      "data": Object {
        "1524759180000": Object {
          "200": 2368,
          "204": 85,
          "301": 182,
        },
        "1524759240000": Object {
          "200": 2797,
          "204": 100,
          "301": 232,
        },
      },
      "loaded": true,
      "loading": false,
      "range": 330,
    },
  },
  "test.service": Object {
    "http_codes": Object {
      "data": Object {
        "1524759180000": Object {
          "200": 2368,
          "204": 85,
          "301": 182,
          "302": 11,
          "304": 1,
          "404": 173,
        },
        "1524759240000": Object {
          "200": 2797,
          "204": 100,
          "301": 232,
          "302": 20,
          "304": 7,
          "404": 206,
        },
      },
      "loaded": true,
      "loading": false,
      "range": 10,
    },
  },
}
`;

exports[`metric reducer should handle START_METRIC_LOADING 1`] = `
Object {
  "test.service": Object {
    "http_codes": Object {
      "loaded": false,
      "loading": true,
      "range": 336,
    },
  },
}
`;

exports[`metric reducer should handle START_METRIC_LOADING and remove data from store 1`] = `
Object {
  "test.second.service": Object {
    "bamboozled_by_app": Object {
      "loaded": false,
      "loading": true,
      "range": 330,
    },
    "bamboozled_by_bro": Object {
      "loaded": false,
      "loading": false,
      "range": 330,
    },
    "http_codes": Object {
      "data": Object {
        "1524759180000": Object {
          "200": 2368,
          "204": 85,
          "301": 182,
        },
        "1524759240000": Object {
          "200": 2797,
          "204": 100,
          "301": 232,
        },
      },
      "loaded": true,
      "loading": false,
      "range": 330,
    },
  },
  "test.service": Object {
    "http_codes": Object {
      "loaded": false,
      "loading": true,
      "range": 336,
    },
  },
}
`;

exports[`metric reducer should return the initial state 1`] = `Object {}`;
